partner_contact_in_several_companies/models/res_partner.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models, _
+from odoo.osv import expression
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    contact_type = fields.Selection(
+        [('standalone', _('Standalone Contact')),
+         ('attached', _('Attached to existing Contact')),
+         ],
+        compute='_compute_contact_type',
+        store=True,
+        index=True,
+        default='standalone')
+    contact_id = fields.Many2one('res.partner', string='Main Contact',
```

Comment:
Please use one style for indentation. Best always indent with four spaces, it makes for much nicer code then having everything crammed at the end of lines.